src/util/storable_builder.rs

```diff
@@ -0,0 +1,71 @@
+use crate::crypto::chacha20poly1305::ChaCha20Poly1305;
+use crate::types::{EncryptionMetadata, PlaintextBlob, Storable};
+use ::prost::Message;
+use rand::rngs::ThreadRng;
+use rand::RngCore;
+use std::borrow::Borrow;
+use std::io;
+use std::io::{Error, ErrorKind};
+
+pub struct StorableBuilder {
```

Comment:
Please add docs to all `pub` types and members. Also, please add compile time enforcement with:

```rust
#![deny(missing_docs)]
```